js/core/src/genkit.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NodeSDKConfiguration } from '@opentelemetry/sdk-node';
+import z from 'zod';
+import {
+  CallableFlow,
+  defineFlow,
+  defineStreamingFlow,
+  Flow,
+  FlowConfig,
+  FlowFn,
+  FlowServer,
+  FlowServerOptions,
+  StreamableFlow,
+  StreamingFlowConfig,
+} from './flow.js';
+import { logger } from './logging.js';
+import { PluginProvider } from './plugin.js';
+import { ReflectionServer } from './reflection.js';
+import * as registry from './registry.js';
+import { AsyncProvider, Registry, runWithRegistry } from './registry.js';
+import {
+  LoggerConfig,
+  TelemetryConfig,
+  TelemetryOptions,
+} from './telemetryTypes.js';
+import { cleanUpTracing, enableTracingAndMetrics } from './tracing.js';
+import { getCurrentEnv, isDevEnv } from './utils.js';
+
+/**
+ * Options for initializing Genkit.
+ */
+export interface GenkitOptions {
+  /** List of plugins to load. */
+  plugins?: PluginProvider[];
+  /** Name of the trace store to use. If not specified, a dev trace store will be used. The trace store must be registered in the config. */
+  traceStore?: string;
+  /** Name of the flow state store to use. If not specified, a dev flow state store will be used. The flow state store must be registered in the config. */
+  flowStateStore?: string;
+  /** Whether to enable tracing and metrics. */
+  enableTracingAndMetrics?: boolean;
+  /** Level at which to log messages.*/
+  logLevel?: 'error' | 'warn' | 'info' | 'debug';
+  /** Directory where dotprompts are stored. */
+  promptDir?: string;
+  /** Telemetry configuration. */
+  telemetry?: TelemetryOptions;
```

Comment:
can you please add a FIXME -- I want to move this outta here... telemetry should not (cannot) be scoped to genkit instance.